### PR TITLE
[CHERI-RISC-V] Add support for C{Get,Set}High

### PR DESCRIPTION
**Pull request by arichardson**
Adds assembly level support and Clang builtins for CGetHigh/CSetHigh.

### CODE REVIEW

**Comment by arichardson** [comment]
@jrtc27 Morello uses `int_cheri_cap_copy_from_high` and `int_cheri_cap_copy_to_high` for the first two - I guess we want to use `int_cheri_cap_high_{get,set}` instead?

**Comment by arichardson** [comment]
For `top` I'm not too sure since it's somewhat ambiguous with "give me the top bits". Same for "upper". - Morello uses CGLIM, maybe we want to use `int_cheri_cap_limit_get` instead of `top_get`?

**Comment by arichardson** [comment]
I've dropped CGetTop from this PR, so it's just C{Get/Set}High now with clang support.

File: llvm/test/CodeGen/CHERI-Generic/Inputs/intrinsics.ll [hunk 1]
```diff
@@ -63,6 +64,13 @@ define iCAPRANGE @address_get(i8 addrspace(200)* %cap) nounwind {
   ret iCAPRANGE %address
 }
 
+;; llvm.cheri.cap.address.get is not supported for MIPS - this could be added by
```
Comment (jrtc27) [inline llvm/test/CodeGen/CHERI-Generic/Inputs/intrinsics.ll:67]: s/address/high/

File: llvm/test/CodeGen/CHERI-Generic/Inputs/intrinsics.ll [hunk 2]
```diff
@@ -119,6 +128,13 @@ define i8 addrspace(200)* @bounds_set_exact(i8 addrspace(200)* %cap, iCAPRANGE %
   ret i8 addrspace(200)* %newcap
 }
 
+;; llvm.cheri.cap.address.set is not supported for MIPS - this could be added by
```
Comment (jrtc27) [inline llvm/test/CodeGen/CHERI-Generic/Inputs/intrinsics.ll:131]: s/address/high/

File: clang/test/CodeGen/cheri/cheriintrin.c [hunk 3]
```diff
@@ -159,6 +166,27 @@ void test(void *__capability cap, char *__capability cap2, __SIZE_TYPE__ i) {
  */
 void test_alignment_builtins(void *__capability cap, __SIZE_TYPE__ align);
 
+// CHECK-LABEL: @test_alignment_builtins(
```
Comment (jrtc27) [inline clang/test/CodeGen/cheri/cheriintrin.c:169]: Huh, this was a bit of an oversight

File: llvm/test/CodeGen/CHERI-Generic/Inputs/intrinsics.ll (hunk 1 above)
Comment (jrtc27) [reply llvm/test/CodeGen/CHERI-Generic/Inputs/intrinsics.ll:67]: Also maybe @IF-MIPS@ these so the generated non-MIPS files look more sensible?

**Pull request merged**